File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string s=strs[0];
        bool isPrefix=false;
         bool isLocalPrefix=false;
        while(!isPrefix){
            for(int i=0;i<strs.size();i++){
                if(i==strs.size()-1 && isLocalPrefix && string(strs[i]).find(s)==0 ){
                    return s;
                }
                if(string(strs[i]).find(s)==0){
                    isLocalPrefix=true;
                }
                else{
                    isLocalPrefix=false;
                    break;
                }
            }
            if(!isLocalPrefix){
                s.pop_back();
            }
        }
        return "";
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        // solve by for each letter we compare all of them yk then we return word based on counter
        int count=0;
        int i=0;
        bool diff=false;
        while(i<strs[0].Length){
            for(int j=1;j<strs.Length;j++){
                if(i>=strs[j].Length)
                    diff=true;
                else if(strs[j][i]!=strs[0][i])
                    diff=true;
            }
            if(!diff)
                count++;
            
            else
                break;
            i++;
        }
        return strs[0].Substring(0,count);
    }
}